agent/consul/connect_ca_endpoint.go

```diff
@@ -168,6 +168,13 @@ func (s *ConnectCA) ConfigurationSet(
 		return nil
 	}
 
+	// If the provider hasn't changed, we need to load the current Provider state
+	// so it can decide if it needs to change resources or not based on the config
+	// change.
+	if args.Config.Provider == config.Provider {
+		args.Config.State = config.State
```

Comment:
Should this do a deep copy?

Reply:
Good question. I guess I should document that in the interface. I'm leaning towards no given that it's a `map[string]string` not a blind pointer to something the Provider is likely to hold in the struct but we can document that assumption. If the provider is going to store a map and return it then _it_ should do the copy. In most cases I suspect `State` method will be building this map based on other actual struct members anyway.